.github/workflows/upload.yml

```diff
@@ -0,0 +1,163 @@
+        run: pip3 install jax jaxlib
+
+      - name: Install KaHyPar
+        run: pip3 install kahypar==1.1.7
+
+      - name: Install PennyLane
+        run: |
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Install Lightning-master
+        run: python -m pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
+
+      - name: Run tests
+        run: python -m pytest tests --cov=pennylane $COVERAGE_FLAGS
+
+  device-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        config:
+          - { device: "default.qubit", shots: None }
+          - { device: "default.qubit", shots: 10000 }
+          # - {device: "default.qubit.tf", shots: None}
+          - { device: "default.qubit.autograd", shots: None }
+          - { device: "default.mixed", shots: None }
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-ci.txt
+          pip install wheel pytest pytest-cov pytest-mock flaky --upgrade
+
+      - name: Conditionally install PyTorch
+        if: contains(matrix.config.device, 'torch')
+        run: pip3 install torch==$TORCH_VERSION -f https://download.pytorch.org/whl/torch_stable.html
+
+      - name: Conditionally install TensorFlow
+        if: contains(matrix.config.device, 'tf')
+        run: pip3 install tensorflow==$TF_VERSION
+
+      - name: Conditionally install Jax
+        if: contains(matrix.config.device, 'jax')
+        run: pip3 install jax jaxlib
+
+      - name: Install PennyLane
+        run: |
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Run tests
+        run: |
+          python -m pytest pennylane/devices/tests \
+            --device=${{ matrix.config.device }} \
+            --shots=${{ matrix.config.shots }} \
+            --cov=pennylane $COVERAGE_FLAGS
+
+
+  upload:
+    runs-on: ubuntu-latest
+    needs: [core-tests, device-tests]
+    if: ${{ always() && !cancelled() && needs.build.result == 'success' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Build and install PennyLane
+        run: |
+          python -m pip install --upgrade pip wheel
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
```

Comment:
Somewhere here we should make sure that we are building the release version of pennylane

Reply:
Ah never mind, I think that is done automatically with the checkout action, see:
```
- uses: actions/checkout@v3
  with:
    # Repository name with owner. For example, actions/checkout
    # Default: ${{ github.repository }}
    repository: ''

    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
    ref: ''
```